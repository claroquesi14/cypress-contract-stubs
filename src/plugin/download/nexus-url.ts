import { configVars, RemoteStub, stubCoordinate } from '../stubs/stubs-config';

/**
 * Nexus Repository Manager 3
 * https://help.sonatype.com/repomanager3/rest-and-integration-api/search-api
 *
 * @param config
 */
export function nexus3Url(config: RemoteStub): URL {
  const { groupId, artifactId, version, classifier } = stubCoordinate(config.id);
  const endpoint = `${configVars.endpointNexus3Context}/${configVars.endpointNexus3}`;

  const url = new URL(endpoint, config.server || configVars.server);
  url.searchParams.append('repository', config.repository || configVars.repository);
  url.searchParams.append('group', groupId);
  url.searchParams.append('name', artifactId);
  url.searchParams.append('maven.classifier', classifier);
  url.searchParams.append('maven.extension', 'jar');

  if (version === '+' || version.toLowerCase() === 'latest') {
    url.searchParams.append('sort', 'version');
  } else {
    url.searchParams.append('version', version);
  }

  return url;
}

/**
 * Nexus Repository Manager 2
 * https://repository.sonatype.org/nexus-restlet1x-plugin/default/docs/path__artifact_maven_content.html
 *
 * @param config
 */
export function nexusUrl(config: RemoteStub): URL {
  const { groupId, artifactId, version, classifier } = stubCoordinate(config.id);
  const endpoint = `${configVars.endpointNexusContext}/${configVars.endpointNexus}`;

  const url = new URL(endpoint, config.server || configVars.server);
  url.searchParams.append('r', config.repository || configVars.repository);
  url.searchParams.append('g', groupId);
  url.searchParams.append('a', artifactId);
  url.searchParams.append('c', classifier);
  url.searchParams.append('p', 'jar');

  if (version === '+' || version.toLowerCase() === 'latest') {
    url.searchParams.append('v', 'LATEST');
  } else {
    url.searchParams.append('v', version);
  }

  return url;
}
