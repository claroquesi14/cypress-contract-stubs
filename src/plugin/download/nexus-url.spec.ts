import { nexus3Url, nexusDownloadUrl, nexusUrl } from './nexus-url';
import { expect } from '@jest/globals';

describe('NexusUrl Test', () => {
  const fixedVersionId = 'test.group:test-artifact:1.2:stubs';
  const latestVersionId = 'test.group:test-artifact:+:stubs';

  describe('when Nexus3 repository manager', () => {
    it('should return the url with fixed version', () => {
      const url = nexus3Url({ id: fixedVersionId });
      expect(url.toString()).toMatchSnapshot();
    });

    it('should return the url with latest version', () => {
      const url = nexus3Url({ id: latestVersionId });
      expect(url.toString()).toMatchSnapshot();
    });
  });

  describe('when Nexus repository manager', () => {
    it('should return the url with fixed version', () => {
      const url = nexusUrl({ id: fixedVersionId });
      expect(url.toString()).toMatchSnapshot();
    });

    it('should return the url with latest version', () => {
      const url = nexusUrl({ id: latestVersionId });
      expect(url.toString()).toMatchSnapshot();
    });

    it('should return the download url', () => {
      const url = nexusDownloadUrl('internal/path/stub.jar', { id: fixedVersionId });
      expect(url.toString()).toMatchSnapshot();
    });
  });
});
