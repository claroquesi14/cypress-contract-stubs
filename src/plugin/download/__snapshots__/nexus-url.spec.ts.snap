// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NexusUrl Test when Nexus repository manager should return the download url 1`] = `"http://localhost:8081/service/local/repositories/maven-releases/content/internal/path/stub.jar"`;

exports[`NexusUrl Test when Nexus repository manager should return the url with fixed version 1`] = `"http://localhost:8081/service/local/artifact/maven/resolve?r=maven-releases&g=test.group&a=test-artifact&c=stubs&p=jar&v=1.2"`;

exports[`NexusUrl Test when Nexus repository manager should return the url with latest version 1`] = `"http://localhost:8081/service/local/artifact/maven/resolve?r=maven-releases&g=test.group&a=test-artifact&c=stubs&p=jar&v=LATEST"`;

exports[`NexusUrl Test when Nexus3 repository manager should return the url with fixed version 1`] = `"http://localhost:8081/service/rest/v1/search/assets?repository=maven-releases&group=test.group&name=test-artifact&maven.classifier=stubs&maven.extension=jar&version=1.2"`;

exports[`NexusUrl Test when Nexus3 repository manager should return the url with latest version 1`] = `"http://localhost:8081/service/rest/v1/search/assets?repository=maven-releases&group=test.group&name=test-artifact&maven.classifier=stubs&maven.extension=jar&sort=version"`;
